.github/workflows/unit-tests.yml

```
@@ -22,39 +23,76 @@ jobs:
-          # Linux clang
-          - { compiler: clang-11, os: ubuntu-20.04, buildType: Debug, cxxStandard: 14, externalSanitizer: true }
-          - { compiler: clang-11, os: ubuntu-20.04, buildType: Debug, cxxStandard: 17 }
-          - { compiler: clang-11, os: ubuntu-20.04, buildType: Release, cxxStandard: 14 }
-          # Linux Latest GCC
-          - { compiler: gcc-12,   os: ubuntu-22.04, buildType: Debug, cxxStandard: 14, coverage: true }
-          # Linux Latest clang
-          - { compiler: clang-16, os: ubuntu-20.04, buildType: Debug, cxxStandard: 14, externalSanitizer: true }
+          # MacOSX:
+          #  Use system clang (14)
+          #   Use (compiler) default C++ standard (14)
+          #   Use system cmake (version gets ignored below)
+          #   Use default boost
+          - { compiler: clang,    os: macos-12,     type: Debug }
+
+          # Linux:
+          #  Oldest Compilers 
+          #   GCC 9 also known to show a few warnings that newer versions dont show
+          #   Use (compiler) default C++ standard (14)
+          #   Use system cmake
+          #   Use system boost (min version)
+          - { compiler: gcc-9,    os: ubuntu-20.04, type: Debug, cmake: 3.16.3, boost: 1.69.0 }
+          - { compiler: clang-10, os: ubuntu-20.04, type: Debug, cmake: 3.16.3, boost: 1.69.0, externalSanitizer: true }
+          #
+          #   Default compiler for Ubuntu 20.04
+          #   Use (compiler) default C++ standard (14)
+          #   Use default cmake
+          #   Use default boost
+          - { compiler: gcc-10,   os: ubuntu-20.04, type: Debug, coverage: true }
+          #
+          #   Default compilers for Ubuntu 22.04
+          #   Use C++ standard (17, default for gcc)
+          #   Use system cmake
+          #   Use system boost
+          - { compiler: gcc-11,   os: ubuntu-22.04, type: Debug, cmake: 3.22.1, boost: 1.74.0 }
+          - { compiler: clang-14, os: ubuntu-22.04, type: Debug, cmake: 3.22.1, boost: 1.74.0, cxx: 17 }
+          #
+          #  Latest Compilers
+          #   GCC 12 needs boost 1.82 to compile correctly
+          #   Use (compiler) default C++ standard (17)
+          #   Use default cmake
+          - { compiler: gcc-12,   os: ubuntu-22.04, type: Debug, boost: 1.82.0 }
+          #   Use default boost
+          - { compiler: clang-16, os: ubuntu-20.04, type: Debug, externalSanitizer: true }
 
     runs-on: ${{matrix.os}}
 
     steps:
       - run: echo "DEPS_DIR=${{runner.temp}}/.cache" >> $GITHUB_ENV
-      - run: echo "CMAKE_VERSION=${{matrix.cmakeVersion}}" >> $GITHUB_ENV
-        if: matrix.cmakeVersion
-      - run: echo "BOOST_VERSION=${{matrix.boostVersion}}" >> $GITHUB_ENV
-        if: matrix.boostVersion
-      - run: echo "ADDITIONAL_CMAKE_FLAGS=${ADDITIONAL_CMAKE_FLAGS} -DRTTR_EXTERNAL_BUILD_TESTING=ON -DRTTR_ENABLE_SANITIZERS=ON" >> $GITHUB_ENV
+
+      - name: "Set BUILD_TYPE"
```

IMO giving this names is overly verbose but fine for me. At least get rid of the quotes (here and below) as in `name: Set C++ standard`:
```suggestion
      - name: Set build type (Release/Debug/...)
```

---

oh there were some with quotes and some without. Ive removed it